odbc/src/api/bind_col_tests.rs

```diff
@@ -0,0 +1,319 @@
+            // values themselves do not matter.
+            let mock_query = &mut MongoQuery::new(
+                vec![doc! {"x": "y"}, doc! {"x": "z"}],
+                vec![
+                    MongoColMetadata::new(
+                        "test_db",
+                        "dn".to_string(),
+                        "fn".to_string(),
+                        Schema::Atomic(Atomic::Scalar(BsonTypeName::Int)),
+                        Nullability::SQL_NO_NULLS,
+                        TypeMode::Simple,
+                    ),
+                    MongoColMetadata::new(
+                        "test_db",
+                        "dn".to_string(),
+                        "fn".to_string(),
+                        Schema::Atomic(Atomic::Scalar(BsonTypeName::Int)),
+                        Nullability::SQL_NO_NULLS,
+                        TypeMode::Simple,
+                    ),
+                ],
+            );
+
+            // Must call next to set the `current` field.
+            let _ = mock_query.next(None);
+
+            // Set the mongo_statement
+            *s.mongo_statement.write().unwrap() = Some(Box::new(mock_query.clone()));
+
+            let indicator: *mut Len = null_mut();
+            let buffer: *mut std::ffi::c_void = Box::into_raw(Box::new([0u8; 4])) as *mut _;
+
+            // Assert that SQLBindCol is successful
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLBindCol(
+                    stmt as *mut _,
+                    1,
+                    CDataType::SQL_C_SLONG as SmallInt,
+                    buffer,
+                    4,
+                    indicator
+                )
+            );
+
+            // Assert that bound_cols has the correct value inside
+            assert_eq!(
+                Some(map! {
+                    1 => BoundColInfo {
+                        target_type: CDataType::SQL_C_SLONG as SmallInt,
+                        target_buffer: buffer,
+                        buffer_length: 4,
+                        length_or_indicator: indicator,
+                    }
+                }),
+                *s.bound_cols.read().unwrap()
+            );
+
+            // free buffer
+            let _ = Box::from_raw(buffer as *mut WChar);
+        }
+    }
+
+    #[test]
+    fn test_unbinding_column() {
+        // Set up MongoHandle
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+
+        unsafe {
+            // Get Statement
+            let s = (*stmt).as_statement().unwrap();
+
+            // Set the bound_cols to non-None initially.
+            *s.bound_cols.write().unwrap() = Some(map! {
```

Comment:
For this test, we might benefit from having `bound_cols` include more than one mapping. That way, the test can show with more certainty that only the referenced column is unbound, as opposed to the whole map being emptied. Would you update this map to include a second binding, and then update the final assertion to show that that second binding still remains even after unbinding `1`?